modules/gapi/src/streaming/onevpl/utils.cpp

```diff
@@ -422,69 +323,65 @@ std::string mfxstatus_to_string(int64_t err) {
-            return "MFX_ERR_DEVICE_LOST";
-        case MFX_ERR_INCOMPATIBLE_VIDEO_PARAM:
-            return "MFX_ERR_INCOMPATIBLE_VIDEO_PARAM";
-        case MFX_ERR_INVALID_VIDEO_PARAM:
-            return "MFX_ERR_INVALID_VIDEO_PARAM";
-        case MFX_ERR_UNDEFINED_BEHAVIOR:
-            return "MFX_ERR_UNDEFINED_BEHAVIOR";
-        case MFX_ERR_DEVICE_FAILED:
-            return "MFX_ERR_DEVICE_FAILED";
-        case MFX_ERR_MORE_BITSTREAM:
-            return "MFX_ERR_MORE_BITSTREAM";
-        case MFX_ERR_GPU_HANG:
-            return "MFX_ERR_GPU_HANG";
-        case MFX_ERR_REALLOC_SURFACE:
-            return "MFX_ERR_REALLOC_SURFACE";
-        case MFX_ERR_RESOURCE_MAPPED:
-            return "MFX_ERR_RESOURCE_MAPPED";
-        case MFX_ERR_NOT_IMPLEMENTED:
-            return "MFX_ERR_NOT_IMPLEMENTED";
-        case MFX_WRN_DEVICE_BUSY:
-            return "MFX_WRN_DEVICE_BUSY";
-        case MFX_WRN_VIDEO_PARAM_CHANGED:
-            return "MFX_WRN_VIDEO_PARAM_CHANGED";
-        case MFX_WRN_IN_EXECUTION:
-            return "MFX_WRN_IN_EXECUTION";
-
-        default:
-            break;
+    switch(err) {
+        ONEVPL_STRINGIFY_CASE(MFX_ERR_NONE)
+        ONEVPL_STRINGIFY_CASE(MFX_ERR_UNKNOWN)
+        ONEVPL_STRINGIFY_CASE(MFX_ERR_NULL_PTR)
+        ONEVPL_STRINGIFY_CASE(MFX_ERR_UNSUPPORTED)
+        ONEVPL_STRINGIFY_CASE(MFX_ERR_MEMORY_ALLOC)
+        ONEVPL_STRINGIFY_CASE(MFX_ERR_NOT_ENOUGH_BUFFER)
+        ONEVPL_STRINGIFY_CASE(MFX_ERR_INVALID_HANDLE)
+        ONEVPL_STRINGIFY_CASE(MFX_ERR_LOCK_MEMORY)
+        ONEVPL_STRINGIFY_CASE(MFX_ERR_NOT_INITIALIZED)
+        ONEVPL_STRINGIFY_CASE(MFX_ERR_NOT_FOUND)
+        ONEVPL_STRINGIFY_CASE(MFX_ERR_MORE_DATA)
+        ONEVPL_STRINGIFY_CASE(MFX_ERR_MORE_SURFACE)
+        ONEVPL_STRINGIFY_CASE(MFX_ERR_ABORTED)
+        ONEVPL_STRINGIFY_CASE(MFX_ERR_DEVICE_LOST)
+        ONEVPL_STRINGIFY_CASE(MFX_ERR_INCOMPATIBLE_VIDEO_PARAM)
+        ONEVPL_STRINGIFY_CASE(MFX_ERR_INVALID_VIDEO_PARAM)
+        ONEVPL_STRINGIFY_CASE(MFX_ERR_UNDEFINED_BEHAVIOR)
+        ONEVPL_STRINGIFY_CASE(MFX_ERR_DEVICE_FAILED)
+        ONEVPL_STRINGIFY_CASE(MFX_ERR_MORE_BITSTREAM)
+        ONEVPL_STRINGIFY_CASE(MFX_ERR_GPU_HANG)
+        ONEVPL_STRINGIFY_CASE(MFX_ERR_REALLOC_SURFACE)
+        ONEVPL_STRINGIFY_CASE(MFX_ERR_RESOURCE_MAPPED)
+        ONEVPL_STRINGIFY_CASE(MFX_ERR_NOT_IMPLEMENTED)
+        ONEVPL_STRINGIFY_CASE(MFX_WRN_DEVICE_BUSY)
+        ONEVPL_STRINGIFY_CASE(MFX_WRN_VIDEO_PARAM_CHANGED)
+        ONEVPL_STRINGIFY_CASE(MFX_WRN_IN_EXECUTION)
+        default: break;
     }
 
     std::string ret("<unknown ");
     ret += std::to_string(err) + ">";
     return ret;
 }
+
+std::string ext_mem_frame_type_to_cstr(int type) {
+    std::stringstream ss;
+    APPEND_STRINGIFY_MASK_N_ERASE(type, "|", MFX_MEMTYPE_DXVA2_DECODER_TARGET);
+    APPEND_STRINGIFY_MASK_N_ERASE(type, "|", MFX_MEMTYPE_DXVA2_PROCESSOR_TARGET);
+    //APPEND_STRINGIFY_MASK_N_ERASE(type, "|", MFX_MEMTYPE_VIDEO_MEMORY_DECODER_TARGET);
```

Comment:
Is it debug code? Need to add comment about these if these a future features.

Reply:
according to docs MFX_MEMTYPE_VIDEO_MEMORY_DECODER_TARGET has the same digit-like value  as  MFX_MEMTYPE_DXVA2_PROCESSOR_TARGET. Need to put comment about this situation